packages/flutter_tools/test/protocol_discovery_test.dart

```diff
@@ -195,6 +195,26 @@ void main() {
         discoverer.cancel();
         logReader.dispose();
       });
+      
+      testUsingContext('ipv6 with Ascii Escape code', () async {
+        final MockDeviceLogReader logReader = new MockDeviceLogReader();
+        final ProtocolDiscovery discoverer = new ProtocolDiscovery.observatory(
+          logReader,
+          portForwarder: new MockPortForwarder(99),
+          hostPort: 54777,
+          ipv6: true,
+        );
+
+        // Get next port future.
+        final Future<Uri> nextUri = discoverer.uri;
+        logReader.addLine('I/flutter : Observatory listening on http://127.0.0.1:54804/PTwjm8Ii8qg=/\x1b[');
```

Comment:
That's not an IPv6 address

Reply:
I'll replace with `http://[::1]:54777/PTwjm8Ii8qg=/` but is not that clear for me how the test above works.
It does the same and it's used for ipv6, I haven't digged if flag `ipv6: true` do some mock replace